issue: portable `libsdl` lib

### Steps

1. Remove the `build` directory
2. Run `cmake -S . -B build`
3. See the logs

``` shell
-- 
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "3.0.2")  
-- Found Freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so (found version "2.11.1") 
CMake Error at CMakeLists.txt:70 (find_package):
  By not providing "FindSDL2_ttf.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "SDL2_ttf",
  but CMake did not find one.

  Could not find a package configuration file provided by "SDL2_ttf" with any
  of the following names:

    SDL2_ttfConfig.cmake
    sdl2_ttf-config.cmake

  Add the installation prefix of "SDL2_ttf" to CMAKE_PREFIX_PATH or set
  "SDL2_ttf_DIR" to a directory containing one of the above files.  If
  "SDL2_ttf" provides a separate development package or SDK, be sure it has
  been installed.
```

### Solution

Making `libsdl` portable and cross-platform by cloning it and building it from its source code

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,10 +64,17 @@ CPMAddPackage(
         "SDL_EXAMPLES OFF"
 )
 
+CPMAddPackage(
+    NAME SDL2_ttf
+    GITHUB_REPOSITORY libsdl-org/SDL_ttf
+    GIT_TAG release-2.22.0
+    OPTIONS
+        "SDL2TTF_SAMPLES OFF"
+)
+
 # OpenSSL / FreeType
 find_package(OpenSSL REQUIRED)
 find_package(Freetype REQUIRED)
-find_package(SDL2_ttf REQUIRED)
 
 # ------------------------------------------------------------------------------
 # Skia (external build)

